org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/variables/CompareObjectsTest.java

```diff
@@ -0,0 +1,527 @@
+			assertEquals("Object contains same elements as of s2 but in different order", resultValue);
+			result.clear();
+
+			result.put(s3, objectComparision.listElementsExtraction((IJavaObject) s3.getValue()));
+			result.put(s4, objectComparision.listElementsExtraction((IJavaObject) s4.getValue()));
+			result = objectComparision.compareSelectedLists(result);
+			compareResult = (Map<String, String>) result.get(s3);
+			o = compareResult.get("Values");
+			resultValue = o.toString();
+			assertEquals("Element is actually missing - [apple1]", resultValue, "[apple1]");
+			result.clear();
+
+			result.put(s5, objectComparision.listElementsExtraction((IJavaObject) s5.getValue()));
+			result.put(s6, objectComparision.listElementsExtraction((IJavaObject) s6.getValue()));
+			result = objectComparision.compareSelectedLists(result);
+			compareResult = (Map<String, String>) result.get(s6);
+			o = compareResult.get("Values");
+			resultValue = o.toString();
+			assertEquals("Element is actually missing - [22]", resultValue, "[22]");
+			result.clear();
+
+			result.put(stack, objectComparision.listElementsExtraction((IJavaObject) stack.getValue()));
+			result.put(arrayList, objectComparision.listElementsExtraction((IJavaObject) arrayList.getValue()));
+			result.put(vector, objectComparision.listElementsExtraction((IJavaObject) vector.getValue()));
+			result.put(linkedList, objectComparision.listElementsExtraction((IJavaObject) linkedList.getValue()));
+			result = objectComparision.compareSelectedLists(result);
+			compareResult = (Map<String, String>) result.get(stack);
+			o = compareResult.get("ImmediateResult");
+			resultValue = o.toString();
+			assertTrue("Same elements in diff order", resultValue.contains("contains same"));
+			compareResult = (Map<String, String>) result.get(vector);
+			o = compareResult.get("ImmediateResult");
+			resultValue = o.toString();
+			assertTrue("Same elements in diff order", resultValue.contains("different order"));
+			compareResult = (Map<String, String>) result.get(arrayList);
+			o = compareResult.get("ImmediateResult");
+			resultValue = o.toString();
+			assertTrue("Same elements in diff order", resultValue.contains("contains same"));
+			compareResult = (Map<String, String>) result.get(linkedList);
+			o = compareResult.get("MultiValues");
+			resultValue = o.toString();
+			assertTrue("Missing elements from others", resultValue.contains("{ArrayList=[Banana]"));
+			result.clear();
+
+			bp.delete();
+		} finally {
+			terminateAndRemove(mainThread);
+			removeAllBreakpoints();
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public void testForMaps() throws Exception { // Test for Maps
+
+		String typeName = "CompreObjects.CompareMapObjects";
+		IJavaLineBreakpoint bp = createLineBreakpoint(37, typeName);
+		IJavaThread mainThread = null;
+		Map<IJavaVariable, Object> result = new HashMap<>();
+		Map<String, String> compareResult = new HashMap<>();
+		String resultKey;
```

Comment:
Delay the variable declarations to it's first initialization if the result is not used outside the code block.